util/py/install/fix_imports/fix_imports.py

```diff
@@ -0,0 +1,63 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+import ctypes
+import os
+import sys
+from pathlib import Path
+
+
+def fix_imports() -> None:
+    """Enable absolute and relative imports in scripts.
+
+    Absolute and relative imports work when a file is executed as a module, e.g.
+    `python3 -m util.foo.bar`, but break when the same file is executed as a script,
+    e.g. `python3 util/foo/bar.py`. In OpenTitan, we would like to structure our python
+    files freely, use relative imports for ease of maintenance, and be able to run them
+    as scripts. This function helps us achieve this goal by updating `sys.path` to
+    include the root of the repo and updating `__package__` in the outer frame. Modules
+    that use absolute or relative imports should call this function if they are imported
+    from a script.
+
+    Typical usage:
+        from fix_imports import fix_imports
+        fix_imports()
+
+    See https://peps.python.org/pep-0328/#relative-imports-and-name.
+    """
+    # Get the module that imports this module. `filename` always points to the actual
+    # file even when run by bazel.
+    frame = sys._getframe(1)
+    filename = Path(frame.f_code.co_filename)
+    # Find the actual repo root using `filename` instead of `os.getcwd()`. The output of
+    # `os.getcwd()` depends whether the script is run by bazel while `filename` is the
+    # actual path, i.e. not a path under `execroot`.
+    repo_root = None
+    # Walk up the tree to avoid additional dependencies.
+    for cand_root in [filename] + [p for p in filename.parents]:
```

Comment:
Nit: Could the list comprehension be replaced with `filename.parents` or `list(filename.parents)`?

Reply:
yes `list(filename.parents)` what I should have used. `filename.parents` doesn't work because it's not a list (would need to chain to this iterable which is an overkill for this IMO).